gpii/node_modules/processReporter/src/processesBridge.js

```diff
@@ -0,0 +1,335 @@
+        if (aProcInfo.command === commandName) {
+            matchingProcs.push(aProcInfo);
+        }
+        return matchingProcs;
+    }, []);
+};
+
+// Return the first process of an array of processes all with the same
+// command name string.  If there are no matching processes, return null.
+gpii.processes.findFirstProcessByCommand =
+function (commandName, procArray) {
+    var commands =
+        gpii.processes.findProcessesByCommand(commandName, procArray);
+    if (commands.length > 0) {
+        return commands[0];
+    }
+    else {
+        return null;
+    }
+};
+
+// Determine if the state of the given process has changed.  Record the
+// new process information in the monitor.  Return boolean.
+gpii.processes.hasStateChanged = function (monitor) {
+    if (!monitor || !monitor.procInfo) {
+        return false;    // nothing sought === nothing changed.
+    }
+    monitor.newProcInfo =
+    gpii.processes.updateProcInfo(monitor.procInfo);
+    return (monitor.procInfo.state !== monitor.newProcInfo.state);
+};
+
+// Utility function to conflate process state values into "Running"
+// (= true) vs. "Not Running" (= false).  Returns boolean.
+gpii.processes.isRunning = function (state) {
+    var result;
+    switch (state) {
+    case "Running":
+    case "Uninterruptible":
+    case "Sleeping":
+    case "Stopped":
+        result = true;
+        break;
+
+    default:
+    case "Zombie":
+    case "NoSuchProcess":
+        result = false;
+        break;
+    }
+    return result;
+};
+
+// Determine if the state of the given process has changed from
+// "not running" to "running" OR from "running" to "not running".
+// Return boolean.
+gpii.processes.hasSwitchedRunState = function (monitor) {
+    if (!monitor || !monitor.procInfo) {
+        // nothing sought === nothing changed.
+        return false;
+    }
+    monitor.newProcInfo =
+        gpii.processes.updateProcInfo(monitor.procInfo);
+    var wasRunning = gpii.processes.isRunning(monitor.procInfo.state);
+    var isRunning =
+        gpii.processes.isRunning(monitor.newProcInfo.state);
+    return (isRunning !== wasRunning);
+};
+
+// Renew the information on a process, or create a new "NoSuchProcces".
+// Returns a new procInfo structure.
+gpii.processes.updateProcInfo = function (procInfo) {
+    var newProcInfo = null;
+    if (procInfo.state === "NoSuchProcess") {
+        newProcInfo =
+            gpii.processes.findFirstProcessByCommand(procInfo.command);
+    }
+    else {
+        newProcInfo = gpii.processes.findProcessByPid(procInfo.pid);
+    }
+    if (newProcInfo === null) {
+        newProcInfo =
+            gpii.processes.initProcInfoNotRunning(procInfo.command);
+    }
+    return newProcInfo;
+};
+
+// Create information on a non-running process, to use to detect when the
+// process starts.
+gpii.processes.initProcInfoNotRunning = function (command) {
+    var process = {};
```

Comment:
Looks like a JSON structure to me - keep a static one, perhaps with fluid.freezeRecursive applied for safety